apps/allocations/app/components/Panel/NewAllocation.js

```diff
@@ -1,41 +1,57 @@
 import PropTypes from 'prop-types'
 import React from 'react'
-import styled from 'styled-components'
-
 import { DropDown, Info } from '@aragon/ui'
+import { OptionsInput, SettingsInput } from '../../../../../shared/ui'
 
 import {
-  DescriptionInput,
+  AddressDropDownOptions,
+  // DescriptionInput,
   Form,
   FormField,
-  OptionsInput,
-  OptionsInputDropdown,
-  SettingsInput,
   InputDropDown,
 } from '../Form'
 
 // TODO: Extract to shared
 const AVAILABLE_TOKENS = ['ETH', 'ANT', 'GIV', 'FTL', '🦄']
 const ALLOCATION_TYPES = ['Informational', 'Token Transfer']
-const PAYOUT_TYPES = ['One-Time', 'Monthly']
+// const PAYOUT_TYPES = ['One-Time', 'Monthly']
 const INITIAL_STATE = {
-  description: '',
-  votingTokens: null,
+  activePayoutOption: 0,
+  addressBookCandidates: [],
+  // TODO: Merge with userInput
+  addressBookInput: { addr: 0, index: 0 },
+  addressError: false,
+  addressSetting: false,
+  allocationDescription: '',
+  allocationError: false,
   allocationType: '',
   allocationTypeIndex: 1,
-  activePayoutOption: 0,
-  payoutType: '',
-  payoutTypeIndex: 0,
-  payoutToken: '',
-  payoutTokenIndex: 0,
   amount: null,
-  allocationError: false,
   balanceSetting: false,
-  addressSetting: false,
-  options: [],
-  optionsInput: { addr: 0, index: 0 },
-  optionsString: [],
-  optionsInputString: { addr: '' },
+  payoutToken: '',
+  payoutTokenIndex: 0,
+  payoutType: '',
+  payoutTypeIndex: 0,
+  userInput: { addr: '' },
+  userInputCandidates: [],
+  votingTokens: null,
+}
+
+const message = {
+  addressError: 'All options must be addresses and cannot be duplicates.',
+  addressSetting: 'Use address book for options',
+  allocationError: 'Amount must be more than zero and less than limit.',
+  balanceSetting: 'Must vote with entire balance',
+  transferWarning:
+    'This will create a Range Vote and after it closes, it will result in a financial transfer.',
+}
+
+const uniqueAddressValidation = (entries, addr) => {
+  const isAddress = /^(0x)?[0-9a-f]{40}$/i.test(addr) // TODO: replace by: web3.isAddress(addr)
```

Comment:
>1\. isAddress, as below,

Solved

2\. DatePicker shows different icon - "group" instead of calendar, IconCalendar changed its content it seems,

Solved

>3\. There is a weird span tag being inserted between checkbox and IssueDesc in Card/Issue - causing the title being shifted to the right (looks like float:right),

Solved

>4\. Submit of Projects Settings disappeared - no Button (or no screen scrolling?),

Solved at 2d5f7304b3080a3f006afecabe0863204a9cd1f4